crates/scmi/src/devices/iio.rs

```diff
@@ -0,0 +1,843 @@
+        }
+        Ok(None)
+    }
+
+    fn read_axis_offset(&self, path: &str) -> Result<Option<i64>, ScmiDeviceError> {
+        self.read_axis_file(path, "offset")
+    }
+
+    fn read_axis_scale(&self, path: &str) -> Result<Option<f64>, ScmiDeviceError> {
+        self.read_axis_file(path, "scale")
+    }
+
+    fn read_axis(&self, axis: &Axis) -> Result<i64, ScmiDeviceError> {
+        let mut value: i64 =
+            read_number_from_file(Path::new(&(String::from(&axis.path) + "_raw")))?.unwrap();
+        let offset: Option<i64> = self.read_axis_offset(&axis.path)?;
+        if let Some(offset_value) = offset {
+            match value.checked_add(offset_value) {
+                Some(new_value) => value = new_value,
+                None => {
+                    error!(
+                        "IIO offset overflow in {}: {} + {}",
+                        &axis.path,
+                        value,
+                        offset.unwrap()
+                    );
+                    return Err(ScmiDeviceError::GenericError);
+                }
+            }
+        }
+        let scale: Option<f64> = self.read_axis_scale(&axis.path)?;
+        if let Some(scale_value) = scale {
+            let exponent_scale = 10.0_f64.powi(i32::from(axis.custom_exponent));
+            value = (value as f64 * (scale_value / exponent_scale)).round() as i64;
+        }
+        Ok(value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::scmi::ScmiDevice;
+
+    use super::*;
+    use std::{
+        assert_eq, fs,
+        path::{Path, PathBuf},
+    };
+
+    fn make_directory(prefix: &str) -> PathBuf {
+        for i in 1..100 {
+            let path = Path::new(".").join(format!("{prefix}{i}"));
+            if fs::create_dir(&path).is_ok() {
+                return path;
+            }
+        }
+        panic!("Couldn't create test directory");
+    }
+
+    struct IIODirectory {
+        path: PathBuf,
+    }
+
+    impl IIODirectory {
+        fn new(files: &[(&str, &str)]) -> IIODirectory {
+            let path = make_directory("_test");
+            let directory = IIODirectory { path };
+            for (file, content) in files.iter() {
+                fs::write(&directory.path.join(file), content).unwrap();
+            }
+            directory
+        }
+    }
+
+    impl Drop for IIODirectory {
+        fn drop(&mut self) {
+            let _ = fs::remove_dir_all(&self.path);
+        }
+    }
+
+    fn directory_path(directory: &IIODirectory) -> String {
+        directory
+            .path
+            .clone()
+            .into_os_string()
+            .into_string()
+            .unwrap()
+    }
+
+    fn device_properties(path: String, channel: String, name: Option<String>) -> DeviceProperties {
+        let mut pairs = vec![
+            (String::from("path"), path),
+            (String::from("channel"), channel),
+        ];
+        if let Some(name) = name {
+            pairs.push((String::from("name"), name));
```

Comment:
```suggestion
            pairs.push(("name".to_owned(), name));
```

Reply:
and same everywhere else, won't comment on further ones

Reply:
Done.